src/cryptsetup/cryptsetup.c

```diff
@@ -715,6 +715,14 @@ static int run(int argc, char *argv[]) {
                                 if (r < 0)
                                         return log_error_errno(r, "Failed to set LUKS data device %s: %m", argv[3]);
                         }
+#ifdef CRYPT_ANY_TOKEN
+                        /* Tokens are available in LUKS2 only, but it is ok to call (and fail) with LUKS1. */
+                        if (!key_file) {
+                                r = crypt_activate_by_token(cd, argv[2], CRYPT_ANY_TOKEN, NULL, flags);
+                                if (r >= 0)
+                                        return 0;
```

Comment:
i think it would make sense to debug log the error here if this fails, i.e. `log_debug_errno(r, "…`, just to keep this debuggable and see the error code...

How recent is the CRYPT_ANY_TOKEN thing? From time to time we should really drop the #ifdeffery and just depend on a new libcryptsetup version. If the concept is a year old or so already we should really drop that I think.

Reply:
I think the failure here should be quiet - if there is no token (most of the headers), it would only produce a lot of noise. If there is a problem with an existitng token, libcryptsetup will produce error through callback anyway.

(I would better see that systemd-cryptsetup allow debugging mode, we produce a lot of messages there to see what's going on, including token processing log here. In recent libcryptsetup (2.1.x) we send all debug output through callback, so only log level need to be set.)

For CRYPT_ANY_TOKEN - it appeared in cryptsetup2.0.0. Actually, I would probably prefer to require libcryptsetup2 here and remove all these #ifdefs... But I guess it is for another PRm if you want it :)

Reply:
log_debug_errno() *is* quiet, except if you turn on debug logging, hence it's always a good idea to place rather too many log_debug() lines everywhere than too few, to make debugging easier.

hmm, we currently call `crypt_set_log_callback()` everywhere to redirect libcryptsetup's logging to systemd's. Is that not enough? Are you saying that libcryptsetup does some internal priority-based filtering already, and given that we are doing that in systemd a second time we should really turn that off? or do i misunderstand you?

to turn on debug logging it's sufficient to set the SYSTEMD_LOG_LEVEL=debug env var for systemd-cryptsetup@.service btw

Reply:
And yeah, i think we should make libcryptsetup 2 a hard dep these days, it was release december 2017, i think that should be enough. but yeah, maybe a separate PR for that